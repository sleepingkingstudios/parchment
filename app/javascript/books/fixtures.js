import { load } from '../fixtures';

/* eslint-disable-next-line import/prefer-default-export */
export const booksData = load('books');
