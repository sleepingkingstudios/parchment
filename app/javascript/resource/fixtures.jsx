import React from 'react';

export const Block = () => (<div />);
export const Form = () => (<div />);
export const Table = () => (<div />);

export const CreatePage = () => (<div />);
export const IndexPage = () => (<div />);
export const LandingPage = () => (<div />);
export const ShowPage = () => (<div />);
export const UpdatePage = () => (<div />);

export const fixtures = {
  create: {
    component: CreatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/create',
    reducer: ((state = { widget: {} }) => state),
  },
  index: {
    component: IndexPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/',
    reducer: ((state = { widgets: [] }) => state),
  },
  show: {
    component: ShowPage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id',
    reducer: ((state = { widget: {} }) => state),
  },
  update: {
    component: UpdatePage,
    exact: true,
    options: { baseUrl: '/widgets', resourceName: 'widgets' },
    path: '/:id/update',
    reducer: ((state = { widget: {} }) => state),
  },
};
