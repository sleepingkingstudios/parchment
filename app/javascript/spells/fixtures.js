/* eslint-disable-next-line import/prefer-default-export */
export const spellsData = [
  {
    id: 'e17bca01-24d3-48d3-a308-099dd6dca373',
    level: 0,
    name: 'Embers',
    school: 'Evocation',
  },
  {
    id: 'e3641970-fc98-4674-b873-5532d89d9f26',
    level: 3,
    name: "Dragon's Breath",
    school: 'Evocation',
  },
  {
    id: 'a74d8c21-607d-4a3f-8cfd-2bbb48ec196a',
    level: 9,
    name: 'Wrath of the Titans',
    school: 'Evocation',
  },
];
