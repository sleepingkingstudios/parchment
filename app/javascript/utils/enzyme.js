import { valueOrDefault } from './object';
import { safeCapitalize } from './string';

export const extractInjectedComponent = (component) => {
  const injected = component.shallow().find('ComponentWithInjectedProps');

  return injected.exists() ? injected : component;
};

export const toHaveFormControl = expect => (received, expectedControl, withProps) => {
  const formControl = received.find(expectedControl);
  const expectedProps = valueOrDefault(withProps, {});

  expect(formControl).toExist();
  expect(formControl).toHaveProp(expectedProps);

  return { pass: true, message: () => '' };
};

export const toHaveFormGroup = expect => (received, expectedGroup, fn) => {
  const formGroup = received.find(expectedGroup);

  expect(formGroup).toExist();

  fn(extractInjectedComponent(formGroup).shallow());

  return { pass: true, message: () => '' };
};

export const toHaveFormInput = expect => (received, props) => {
  const {
    form,
    inputType,
    prop,
  } = props;

  expect(received).toHaveProp({ prop });

  const formInput = received.find(`${safeCapitalize(prop)}Input`);

  expect(formInput).toExist();
  expect(formInput).toHaveProp({ form });

  const shallowInput = formInput.shallow();

  expect(shallowInput).toExist();
  expect(shallowInput).toHaveDisplayName(valueOrDefault(inputType, 'FormInput'));

  return { pass: true, message: () => '' };
};
