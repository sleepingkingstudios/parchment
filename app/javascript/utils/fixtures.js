/* eslint-disable-next-line import/prefer-default-export */
export const trimMultilineLiteral = (text) => {
  if (!text) { return ''; }

  const paragraphs = text.split('\n\n');

  return paragraphs
    .map((paragraph) => {
      const lines = paragraph.split('\n');

      return lines.map(line => line.trim()).join(' ');
    })
    .join('\n\n');
};
