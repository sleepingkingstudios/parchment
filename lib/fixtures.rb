# frozen_string_literal: true

# Namespace for fixtures, which handle reading data from a serialized source and
# hydrating the database.
module Fixtures
  def self.build(record_class, count: nil, environment: 'fixtures')
    Builder.new(record_class, environment: environment).build(count: count)
  end

  def self.create(record_class, count: nil, environment: 'fixtures')
    Builder.new(record_class, environment: environment).create(count: count)
  end

  def self.read(record_class, count: nil, environment: 'fixtures')
    Builder.new(record_class, environment: environment).read(count: count)
  end

  autoload :Builder, 'fixtures/builder'
end
