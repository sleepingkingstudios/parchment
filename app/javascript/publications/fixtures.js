import { load } from '../fixtures';

const isOfficial = (publication) => {
  const { playtest, publisherName } = publication;

  return !playtest && publisherName === 'Wizards of the Coast';
};

const rawData = load('publications');

/* eslint-disable-next-line import/prefer-default-export */
export const publicationsData = rawData
  .map(publication => Object.assign(
    {},
    publication,
    {
      official: isOfficial(publication),
    },
  ));
