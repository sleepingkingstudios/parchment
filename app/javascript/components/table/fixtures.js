import { humanizeList } from '../../utils/array';

export const formatCrewCapacity = (item) => {
  const count = item.crewCapacity;

  if (count === 1) { return '1 Kerbal'; }

  return `${count} Kerbals`;
};

export const formatExperiments = (item) => {
  humanizeList(item.experiments, { default: 'None' });
};

export const columns = [
  {
    label: 'Name',
    prop: 'name',
  },
  {
    label: 'Crew Capacity',
    prop: 'crewCapacity',
    value: formatCrewCapacity,
  },
  {
    label: 'Experiments',
    prop: 'experiments',
  },
];

export const rocketsData = [
  {
    id: '83f5e9e8-92df-4203-bc2e-4381370c2186',
    name: 'Apprentice',
    crewCapacity: 1,
    experiments: [
      'Crew Report',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '4bd7454c-3b15-4047-8d57-dfa831dac58e',
    name: 'Sorcerer',
    crewCapacity: 2,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
    ],
  },
  {
    id: '2fead083-d9ea-4f04-949e-d8c00e800cbb',
    name: 'Warlock',
    crewCapacity: 3,
    experiments: [
      'Crew Report',
      'Materials Study',
      'Mystery Goo',
      'Temperature Scan',
      'Pressure Scan',
      'Gravity Scan',
    ],
  },
];
