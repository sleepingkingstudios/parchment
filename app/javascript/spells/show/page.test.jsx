import React from 'react';
import { shallow } from 'enzyme';

import ShowSpellPage from './page';
import { spellsData } from '../fixtures';
import {
  INITIALIZED,
  PENDING,
  FAILURE,
  SUCCESS,
} from '../../store/requestStatus';

describe('<ShowSpellPage />', () => {
  const defaultProps = {
    spell: {},
    status: INITIALIZED,
  };
  const breadcrumbs = [
    {
      label: 'Home',
      url: '/',
    },
    {
      label: 'Spells',
      url: '/spells',
    },
    {
      label: 'Loading...',
      active: true,
    },
  ];

  it('should wrap the contents in a Page', () => {
    const rendered = shallow(<ShowSpellPage {...defaultProps} />);

    expect(rendered).toHaveDisplayName('Page');
    expect(rendered).toHaveClassName('page-spells');
    expect(rendered).toHaveProp('breadcrumbs', breadcrumbs);
  });

  it('should not render a spell block', () => {
    const rendered = shallow(<ShowSpellPage {...defaultProps} />);

    expect(rendered.find('SpellBlock')).not.toExist();
  });

  it('should show the loading message', () => {
    const rendered = shallow(<ShowSpellPage {...defaultProps} />);
    const message = 'Loading spell from the server...';

    expect(rendered.find('Page').shallow()).toIncludeText(message);
  });

  it('should match the snapshot', () => {
    const rendered = shallow(<ShowSpellPage {...defaultProps} />);

    expect(rendered).toMatchSnapshot();
  });

  describe('when the request status is PENDING', () => {
    const props = { ...defaultProps, status: PENDING };

    it('should not render a spell block', () => {
      const rendered = shallow(<ShowSpellPage {...props} />);

      expect(rendered.find('SpellBlock')).not.toExist();
    });

    it('should show the loading message', () => {
      const rendered = shallow(<ShowSpellPage {...props} />);
      const message = 'Loading spell from the server...';

      expect(rendered.find('Page').shallow()).toIncludeText(message);
    });
  });

  describe('when the request status is FAILURE', () => {
    const props = { ...defaultProps, status: FAILURE };

    it('should not render a spell block', () => {
      const rendered = shallow(<ShowSpellPage {...props} />);

      expect(rendered.find('SpellBlock')).not.toExist();
    });

    it('should show the failure message', () => {
      const rendered = shallow(<ShowSpellPage {...props} />);
      const message = 'Spell not found.';

      expect(rendered.find('Page').shallow()).toIncludeText(message);
    });
  });

  describe('when the request status is SUCCESS', () => {
    const spell = spellsData[0];
    const props = { ...defaultProps, status: SUCCESS, spell };
    const successBreadcrumbs = [
      {
        label: 'Home',
        url: '/',
      },
      {
        label: 'Spells',
        url: '/spells',
      },
      {
        label: spell.name,
        active: true,
      },
    ];

    it('should update the breadcrumbs', () => {
      const rendered = shallow(<ShowSpellPage {...props} />);

      expect(rendered).toHaveProp('breadcrumbs', successBreadcrumbs);
    });

    it('should render the spell block', () => {
      const rendered = shallow(<ShowSpellPage {...props} />);
      const block = rendered.find('SpellBlock');

      expect(block).toExist();
      expect(block).toHaveProp('spell', spell);
    });

    it('should match the snapshot', () => {
      const rendered = shallow(<ShowSpellPage {...props} />);

      expect(rendered).toMatchSnapshot();
    });
  });
});
