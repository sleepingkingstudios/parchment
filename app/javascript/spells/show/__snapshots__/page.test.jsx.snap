// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowSpellPage /> should match the snapshot 1`] = `
<Page
  breadcrumbs={
    Array [
      Object {
        "label": "Home",
        "url": "/",
      },
      Object {
        "label": "Spells",
        "url": "/spells",
      },
      Object {
        "active": true,
        "label": "Loading...",
      },
    ]
  }
  className="page-spells"
  layout={null}
  subtitle="5e Campaign Companion"
  title="Parchment"
>
  <HeadingWithButtons
    buttons={
      Array [
        Object {
          "label": "Update Spell",
          "outline": true,
          "url": "/spells/undefined/update",
        },
      ]
    }
  >
    Show Spell
  </HeadingWithButtons>
  <p>
    Loading spell from the server...
  </p>
</Page>
`;

exports[`<ShowSpellPage /> when the request status is SUCCESS should match the snapshot 1`] = `
<Page
  breadcrumbs={
    Array [
      Object {
        "label": "Home",
        "url": "/",
      },
      Object {
        "label": "Spells",
        "url": "/spells",
      },
      Object {
        "active": true,
        "label": "Thunderclap",
      },
    ]
  }
  className="page-spells"
  layout={null}
  subtitle="5e Campaign Companion"
  title="Parchment"
>
  <HeadingWithButtons
    buttons={
      Array [
        Object {
          "label": "Update Spell",
          "outline": true,
          "url": "/spells/e17bca01-24d3-48d3-a308-099dd6dca373/update",
        },
      ]
    }
  >
    Show Spell
  </HeadingWithButtons>
  <SpellBlock
    spell={
      Object {
        "castingTime": "1 reaction, which you take when you speak your own name or hear it spoken",
        "description": "Immediately after speaking or hearing your name, you create the illusory sound of an ominous, distant thunderclap.",
        "duration": "Instantaneous",
        "id": "e17bca01-24d3-48d3-a308-099dd6dca373",
        "level": 0,
        "materialComponent": "",
        "name": "Thunderclap",
        "range": "Self",
        "ritual": false,
        "school": "illusion",
        "shortDescription": "Create an ominous thunderclap when your name is spoken",
        "somaticComponent": false,
        "verbalComponent": false,
      }
    }
  />
</Page>
`;
